With.Fody/ModuleWeaver.cs

```diff
@@ -20,52 +21,57 @@ public ModuleWeaver()
 
     public void Execute()
     {
-        foreach (var type in ModuleDefinition.Types.Where(CanHaveWith))
+        foreach (var type in ModuleDefinition.Types)
         {
-            AddWith(type);
-            RemoveGenericWith(type);
-            LogInfo($"Added method 'With' to type '{type.Name}'.");
+            if (type.GetMethods().Any(m => m.IsPublic && m.Name.StartsWith("With")))
```

Comment:
Small thing, but I prefer `.Where` inside `foreach` to nested `if`. Saves you some indentation and reduces visual clutter.